Publish technical documentation to the website repository (#227)
grafana/scenes: docs/sources/scene-objects/scene-app/scene-app-drilldown.md

---
id: scene-app-drilldown
title: Drill-down pages in Scenes apps
menuTitle: Drill-down pages
hide_title: true
weight: 300
---

# Drill-down pages in Scenes apps

Drill-down pages are a powerful tool for building complex, data-driven applications. They allow you to create a high-level overview of the data that a user can interact with and progressively explore to reveal underlying data.

## Add drill-down pages to Scenes apps

`SceneAppPage` comes with an API that allows you to create deep, nested drill-down pages.

> > **Note:** **Before you begin**: You must already know about React Router URL params, Grafana field configuration, and data links before continuing with this guide.

Reviewer: ```suggestion
> **Note:** Before you begin, you must already know about React Router URL params, Grafana field configuration, and data links before continuing with this guide.
```

```ts
function getOverviewScene() {
  const queryRunner = new SceneQueryRunner({
    $timeRange: new SceneTimeRange(),
    datasource: {
      type: 'prometheus',
      uid: '<PROVIDE_GRAFANA_DS_UID>',
    },
    queries: [
      {
        refId: 'A',
        expr: 'sort_desc(avg by(handler) (rate(prometheus_http_request_duration_seconds_sum {}[5m]) * 1e3))',
        format: 'table',
        instant: true,
      },
    ],
  });

  const tablePanel = new VizPanel({
    $data: queryRunner,
    title: 'Average duration of HTTP request',
    pluginId: 'table',
  });

  return new EmbeddedScene({
    body: new SceneFlexLayout({
      direction: 'column',
      children: [
        new SceneFlexItem({
          minHeight: 300,
          body: tablePanel,
        }),
      ],
    }),
  });
}

function getSceneApp() {
  return new SceneApp({
    pages: [
      new SceneAppPage({
        title: 'HTTP handlers overview',
        url: '/a/<PLUGIN_ID>/my-app',
        getScene: getOverviewScene,
      }),
    ],
  });
}
```

### Step 3. Set up drill-down navigation

> To show the drill-down page, you need to provide navigation. Configure Table panel data links (learn about data links in the [official Grafana documentation](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/)). Then modify the Table panel configuration to set up a data link for the `handler` field:

Reviewer: ```suggestion
To show the drill-down page, you need to provide navigation. First, configure Table panel [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/), then modify the Table panel configuration to set up a data link for the `handler` field:
```
